mitodl/mit-open
# podcast search

# What are the relevant tickets?
closes https://github.com/mitodl/mit-open/issues/178

# Description (What does it do?)
This PR adds podcasts and podcast episodes to the learning resources search index. It also refactors the indexing functions to be less repetitive.

# Screenshots (if appropriate):


# How can this be tested?
Run 
`docker-compose run web ./manage.py update_index --all`
`docker-compose run web ./manage.py  backpopulate_podcast_data`
Verify that there are no errors and you see data at 
http://localhost:8063/api/v1/learning_resources_search/?resource_type=podcast_episode
http://localhost:8063/api/v1/learning_resources_search/?resource_type=podcast

Run 
`docker-compose run web ./manage.py  backpopulate_podcast_data --delete`
Verify that there are no results at
http://localhost:8063/api/v1/learning_resources_search/?resource_type=podcast_episode
http://localhost:8063/api/v1/learning_resources_search/?resource_type=podcast

Also verify that refactor did not break course/course file/program index
`docker-compose run web ./manage.py backpopulate_ocw_data --course-name 21w-747-rhetoric-spring-2005`
Should run without errors and make 21w.747 appear in http://localhost:8063/api/v1/learning_resources_search/?q=rhetoric
`docker-compose run web ./manage.py backpopulate_ocw_data --course-name 21w-747-rhetoric-spring-2005 --delete`
should remove the course from the search index

Similarly, `docker-compose run web ./manage.py backpopulate_xpro_data` and  `docker-compose run web ./manage.py backpopulate_xpro_data --delete` should make xpro courses appear/be removed from the search results

# Additional Context
